cmd/snap/main_test.go

```diff
@@ -142,33 +149,26 @@ func (s *SnapSuite) TestVersionOnClassic(c *C) {
 	s.RedirectClientToTestServer(func(w http.ResponseWriter, r *http.Request) {
 		fmt.Fprintln(w, `{"type":"sync","status-code":200,"status":"OK","result":{"on-classic":true,"os-release":{"id":"ubuntu","version-id":"12.34"},"series":"56","version":"7.89"}}`)
 	})
-	origArgs := os.Args
-	defer func() { os.Args = origArgs }()
-	os.Args = []string{"snap", "--version"}
-	err := snap.RunMain()
-	c.Assert(err, IsNil)
-	c.Assert(s.Stdout(), Equals,
-		`snap    7.89
-snapd   7.89
-series  56
-ubuntu  12.34
-`)
+	restore := mockArgs("snap", "--version")
+	defer restore()
+	restore = mockVersion("4.56")
```

Comment:
Ah! I see. Man I feel so stupid now, having written those tests in the first place